src/main/java/java/lang/StringBuilder.java

```diff
@@ -241,23 +241,25 @@ public StringBuilder append(CharSequence s, int start, int end) {
      */
     @Override
     public StringBuilder append(char[] str) {
-        // DIFFBLUE MODEL LIBRARY: Handled internally by CBMC
+        // DIFFBLUE MODEL LIBRARY
+        String string = CProverString.ofCharArray(str, 0, str.length);
+        return append(string);
         // super.append(str);
         // return this;
-        return CProver.nondetWithNullForNotModelled();
     }
 
     /**
      * @throws IndexOutOfBoundsException {@inheritDoc}
-     *
-     * @diffblue.noSupport
+     * @diffblue.fullSupport
+     * @diffblue.untested
```

Comment:
How is this going to be tested?

Reply:
It's tested by this PR https://github.com/diffblue/cbmc/pull/2374 but maybe we should have more tests in the models library.

Reply:
It's tested by this PR https://github.com/diffblue/cbmc/pull/2374

Reply:
but maybe we should add test in the models library